tests/test_binary.nim

```diff
@@ -7,10 +7,12 @@ const
   binaryName = "configlet" & binaryExt
 
 proc main =
-  let repoRootDir = getAppDir().parentDir()
+  let repoRootDir = getCurrentDir()
```

Comment:
If I ran `nim r --styleCheck:error ./tests/all_tests.nim`, this would equal: `/home/erik/.cache/nim`

With this change, it resolved to the directory in which the `nim` command is ran, and running it like above correctly builds the binary.

I'm running Nim 1.4.2, not sure if that matters.

Reply:
> If I ran `nim r --styleCheck:error ./tests/all_tests.nim`, this would equal: `/home/erik/.cache/nim`

Great catch. We should indeed not determine `repoRootDir` from the path of the output binary. The `getAppDir` approach also breaks if somebody writes something like `nim c --outDir:/tmp -r tests/all_tests.nim`. 

> Is it common to run tests from the root directory, or more common to do that from the tests directory?

I don't know, but I think it's best if they can be run from anywhere. So I think `currentSourcePath` is better than `getCurrentDir`.

> I'm running Nim 1.4.2, not sure if that matters.

It doesn't matter, except insofar that `nim r` was only introduced in 1.4.0. But Nim 1.4.0 was released 2 weeks before this code was committed.

Reply:
I also think `currentSourcePath` is indeed the best approach.